File: R/linear_regression.r
DATA_PATH = "/Users/hilsts/Documents/linear-models-benchmark/data/"
df <- read.csv(paste(DATA_PATH,"song_data.csv", sep = ""))

set.seed(100)

df$song_name <- NULL

index = sample(1:nrow(df), 0.75 * nrow(df))

train = df[index, ]
test = df[-index, ]

linear_regression <- function(train, test) {
    lr = lm(song_popularity~., data = train)
    pred1 <- predict(lr, newdata = test)
}

start_time <- Sys.time()
linear_regression(train, test)
end_time <- Sys.time()
print(end_time - start_time)

File: Julia/linear_regression.jl
using CSV
using DataFrames
using Lathe.preprocess: TrainTestSplit
using CSV
using DataFrames
using MLJ
using GLM
using Lasso
using MLBase




df = DataFrame(CSV.File("/Users/hilsts/Documents/linear-models-benchmark/data/song_data.csv"))
df = select!(df, Not(:song_name))
train, test = TrainTestSplit(df,.75)

function bench_linear_regression(train,test)

    linearRegressor = lm(Term(:song_popularity) ~ sum(Term.(Symbol.(names(train[:, Not(:song_popularity)])))), train)
    predict(linearRegressor, test)
    
end

@time bench_linear_regression(train, test)


